platform_plugin_turnitin/turnitin_client/handlers.py

```diff
@@ -0,0 +1,280 @@
+    method_map = {
+        "get": requests.get,
+        "post": requests.post,
+        "put": requests.put,
+        "delete": requests.delete,
+        "patch": requests.patch,
+    }
+
+    # Retrieving the correct function
+    method_func = method_map.get(request_method.lower())
+
+    if not method_func:
+        raise ValueError(f"Unsupported request method: {request_method}")
+
+    # Calling the appropriate method
+    if request_method.lower() in ["post", "put", "patch"]:
+        response = method_func(
+            f"{TII_API_URL}/api/v1/{url_prefix}", headers=headers, json=data
+        )
+    else:
+        response = method_func(
+            f"{TII_API_URL}/api/v1/{url_prefix}", headers=headers, params=data
+        )
+
+    return response
+
+
+def pretty_print_response(response, type_of=""):
+    """debug helper function"""
+    content = response.json()
+    print("\n\n")
+    print(f"------{type_of}------")
+    print("\n\n")
+    print(json.dumps(content, indent=4))
+    print("\n\n")
+    print("------------")
+    print("\n\n")
+
+
+# Returns all the features enabled in Turnitin account.
+def get_features_enabled():
+    """
+    Returns all the features enabled in the Turnitin account.
+    """
+    response = turnitin_api_handler("get", "features-enabled")
+    pretty_print_response(response)
+
+
+# *---EULA endpoints ---*
+# The EULA is a page of terms and conditions that the owner and the
+# submiter has to accept in order to send a file to Turnitin.
+
+
+def get_eula_version_info(version: str = "latest", language: str = "EN"):
+    """
+    Returns Turnitin's EULA (End User License Agreement) version information.
+    The EULA is a page of terms and conditions that both the owner and the submitter
+    have to accept in order to send a file to Turnitin.
+    """
+    response = turnitin_api_handler("get", f"eula/{version}?lang={language}")
+    pretty_print_response(response)
+
+
+def get_eula_page(version: str = "v1beta", language: str = "en-US"):
+    """
+    Returns the HTML content for a specified EULA version.
+    """
+    response = turnitin_api_handler("get", f"/eula/{version}/view?lang={language}")
+    return response
+
+
+def post_accept_eula_version(payload, version: str = "v1beta"):
+    """
+    Accepts a specific EULA version.
+    This method should be invoked after the user has viewed the EULA content.
+    """
+    response = turnitin_api_handler("post", f"eula/{version}/accept", payload)
+    pretty_print_response(response, "ACCEPT EULA")
+    return response
+
+
+def get_eula_acceptance_by_user(user_id):
+    """
+    Checks if a specific user has accepted a particular EULA version.
+    """
+    response = turnitin_api_handler("get", f"eula/v1beta/accept/{user_id}")
+    pretty_print_response(response)
+
+
+# *---Submissions endpoints ---*
+# Submissions is a Turnithin model that has all relative info to a Assessment send by
+# an student.
```

Comment:
Maybe it'd be better to create a package called: /handlers so we have modules for each of these interactions:
/handlers
  submissions.py
  eula.py
  utils.py
It'd read better than separating them with in-line comments. We should use them (in-line comments) when explaining why we're doing something, not to describe it.

Reply:
Agree :)